### PR TITLE
Add initial nix flags impl

### PR DESCRIPTION
**Pull request by yavko**
Pretty self explanatory

### CODE REVIEW

**Comment by yavko** [comment]
format failed?

**Comment by JakeStanger** [comment]
Yeah the format is my fault, that's failing on master. If nix is building that's all that's relevant here so I'm happy with that. Will review later

**Comment by JakeStanger** [comment]
Looks good all except that minor issue above, thanks for getting these in :). 

Btw, if you use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) you'll automatically get included in the changelog.

**Comment by yavko** [comment]
> Looks good all except that minor issue above, thanks for getting these in :). 
> 
> Btw, if you use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) you'll automatically get included in the changelog.

Oh, sorry idk conventional commits, or how to edit commit names

**Comment by yavko** [comment]
Also ill make some changes later today to make it more future proof

**Comment by JakeStanger** [comment]
> Oh, sorry idk conventional commits, or how to edit commit names

It's not a worry as I'm happy to merge it anyway since this won't functionally affect the bar, just going forwards if you can. If you look at the commit history you'll see all mine start with a prefix like `feat:` or `fix:` to indicate the change type - basically just that.

> Also ill make some changes later today to make it more future proof

Okay cool, cheers! I'll hold off merging for now and let me know when you want me to re-review

File: README.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ Here is an example nix flake that uses Ironbar.
             enable = true;
             config = {};
             style = "";
+						package = inputs.ironbar.packageBuilder ["features"];
```
[the remark this answers is not in the document]
Comment (yavko) [reply README.md:62]: on it

**Comment by yavko** [comment]
> Looks good all except that minor issue above, thanks for getting these in :).
> 
> Btw, if you use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) you'll automatically get included in the changelog.

Okay I think I'm done

**Comment by JakeStanger** [comment]
Thanks for the updates, that looks like a more robust solution :)

Last few teething points before I merge:

- It looks like the indentation problem (from my initial review) is still outstanding. Are you able to sort that pls?
- Thanks for updating the commit messages- could you change the latest one pls, just so it's clear it's related to nix? `feat(nix): use cargo default features` would be perfect.

**Comment by yavko** [comment]
> Thanks for the updates, that looks like a more robust solution :)
> 
> Last few teething points before I merge:
> 
>     * It looks like the indentation problem (from my initial review) is still outstanding. Are you able to sort that pls?
> 
>     * Thanks for updating the commit messages- could you change the latest one pls, just so it's clear it's related to nix? `feat(nix): use cargo default features` would be perfect.

Ah okay!

**Comment by yavko** [comment]
Wait I didn't fix properly

**Comment by yavko** [comment]
Now it should be done

**Comment by yavko** [comment]
build seems to be failing for strange reason
image

**Comment by JakeStanger** [comment]
Yeah, looks like something's gone wrong with the Cachix step and it's not installing right. Master has started failing for the same reason. I know this was passing though & you've changed very little since then so I'm confident it's fine.

Thanks for the help!

**Pull request merged**